### PR TITLE
[ALOY-1476] [ALOY-1475] Fix and improve tests against known good code

### PR DESCRIPTION
**Pull request by FokkeZB**
This PR has two purposes:

1. Improve compile tests by showing actual diff from known good code (https://jira.appcelerator.org/browse/ALOY-1475)
2. Fix that Travis did not test known good code (https://jira.appcelerator.org/browse/ALOY-1476)
3. Fix the broken 355 tests :(

### CODE REVIEW

**Comment by FokkeZB** [comment]
So, Travis doesn't fail because the tests failing only run on Darwin. Pushed new commit that includes osx on Travis.

File: .travis.yml [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
-node_js:
-- '0.10'
-- '0.12'
-before_script:
-- npm install -g .
-- npm install -g jake
-- npm install -g titanium
-- mkdir ~/.titanium
-- cp ./cli_config.json ~/.titanium/config.json
+language: cpp
```
Comment (feons) [inline .travis.yml]: Why is this `cpp`?

File: test/lib/testUtils.js [hunk 2]
```diff
@@ -169,7 +170,22 @@ exports.addMatchers = function() {
 			},
 			toBeObject: function(expected) {
 				return _.isObject(this.actual);
+			},
+			toNotDiff: function(expected) {
+				var pass = this.actual === expected;
+				this.message = function() {
+		      return ["Expected to have no diff, but it does: \n\n" + JsDiff.diffLines(expected, this.actual).map(function(part) {
```
Comment (feons) [inline test/lib/testUtils.js]: Weird indentation.

File: .travis.yml (hunk 1 above)
Comment (feons) [inline .travis.yml]: Okay so this is the reason: https://github.com/travis-ci/travis-ci/issues/2311

**Comment by feons** [comment]
Looks good! **Approved.**

**Pull request merged**